kinesis.go

```diff
@@ -60,7 +55,7 @@ func (a *KinesisAdapter) Stream(logstream chan *router.Message) {
 		}
 
 		if d, ok := a.findDrainer(streamName); ok {
-			logErr(d.Buffer.Add(m))
+			go d.Buffer.Add(m)
```

Comment:
Spinning up a new goroutine like this for every message might be potentially dangerous:

1. We won't have control over applying any backpressure so if Add starts falling behind, this could spin up a large number of goroutines.
2. I _think_ there's no guarantee that the goroutines will be executed in the same order that they're started, so log lines might get added out of order.

I wonder if it would be better to treat individual buffers as short lived, where you add messages to the buffer until it's full, then send that buffer off on a channel to get flushed by another goroutine, and allocate a new buffer for the next messages. This is basically what we do in the events service: https://github.com/remind101/r101-events/blob/master/handlers/redshift/receiver.go#L89-L91

It's potentially more allocations, but probably not enough to worry about it.